content/token22-metadata.md

```diff
@@ -0,0 +1,1083 @@
+
+To determine all of this programmatically, we use the `getMintLen` and `pack` functions from the `@solana/spl-token` library:
+
+```ts
+
+const metadata: TokenMetadata = {
+  mint: mint.publicKey,
+  name: tokenName,
+  symbol: tokenSymbol,
+  uri: tokenUri,
+  additionalMetadata: [['customField', 'customValue']],
+};
+
+const mintAndPointerLen = getMintLen([ExtensionType.MetadataPointer]); // Metadata extension is variable length, so we calculate it below
+const metadataLen = TYPE_SIZE + LENGTH_SIZE + pack(metadata).length;
+const totalLen = mintLen + mintAndPointerLen
+const lamports = await connection.getMinimumBalanceForRentExemption(totalLen);
+```
+
+To actually create and initialize the `mint` with the metadata and metadata pointer, we need several instructions in a particular order:
+
+1. Create the `mint` account which reserves space on the blockchain with `SystemProgram.createAccount`
+2. Initialize the metadata pointer extension with `createInitializeMetadataPointerInstruction`
+3. Initialize the mint itself with `createInitializeMintInstruction`
+4. Initialize the metadata with `createInitializeInstruction` (this ONLY sets the basic metadata fields)
+5. Optional: Set the custom fields with `createUpdateFieldInstruction` (one field per call)
+
+```ts
+  const createMintAccountInstructions = SystemProgram.createAccount({
+    fromPubkey: payer.publicKey,
+    lamports,
+    newAccountPubkey: mint.publicKey,
+    programId: TOKEN_2022_PROGRAM_ID,
+    space: totalLen,
+  });
+
+  const initMetadataPointerInstructions = createInitializeMetadataPointerInstruction(
+    mint.publicKey,
+    payer.publicKey,
+    mint.publicKey, // we will point to the mint it self as the metadata account
+    TOKEN_2022_PROGRAM_ID,
+  );
+
+  const initMintInstructions = createInitializeMintInstruction(
+    mint.publicKey,
+    decimals,
+    payer.publicKey,
+    payer.publicKey,
+    TOKEN_2022_PROGRAM_ID,
+  );
+
+  const initMetadataInstruction = createInitializeInstruction({
+    programId: TOKEN_2022_PROGRAM_ID,
+    mint: mint.publicKey,
+    metadata: mint.publicKey,
+    name: metadata.name,
+    symbol: metadata.symbol,
+    uri: metadata.uri,
+    mintAuthority: payer.publicKey,
+    updateAuthority: payer.publicKey,
+  });
+
+  const updateMetadataFieldInstructions = createUpdateFieldInstruction({
+    metadata: mint.publicKey,
+    updateAuthority: payer.publicKey,
+    programId: TOKEN_2022_PROGRAM_ID,
+    field: metadata.additionalMetadata[0][0],
+    value: metadata.additionalMetadata[0][1],
+  });
+  ```
+
+Wrap all of these instructions in a transaction to create the embedded NFT:
+```ts
+const transaction = new Transaction().add(
+  createMintAccountInstructions,
+  initMetadataPointerInstructions,
+  initMintInstructions,
+  initMetadataInstruction,
+  updateMetadataFieldInstructions, // if you want to add any custom field
+);
+const sig = await sendAndConfirmTransaction(connection, transaction, [payer, mint]);
```

Comment:
Just `signature`. There's no reason to contract anything, you're using full words elsewhere and it's clearer.

Reply:
agreed